stl/inc/yvals.h

```diff
@@ -181,35 +181,15 @@ _STL_DISABLE_CLANG_WARNINGS
 #endif // ^^^ !defined(_DEBUG) ^^^
 #endif // !defined(_STL_CRT_SECURE_INVALID_PARAMETER)
 
-#define _STL_REPORT_ERROR(mesg)                  \
-    do {                                         \
-        _RPTF0(_CRT_ASSERT, mesg);               \
-        _STL_CRT_SECURE_INVALID_PARAMETER(mesg); \
-    } while (false)

```

Comment:
I'm a bit lost as to why we had a `do { .. } while (false)` in the first place (it will never loop, after all). Either I'm missing something basic, or there's some wizardry at play. I'll reach out internally.

Reply:
Yep, that's it exactly. I mentioned this in the PR description (very tersely):

> We don't need this because we ALWAYS use braces for control flow.

I think the `do { ... } while (false)` might actually have been introduced during an earlier era where our formatting was radically different and did contain unbraced control flow. Fortunately, this is one of the syntax things that I feel the strongest about (it led to a major bug in the real world!), and we converted over the entire codebase before open-sourcing.